File: week-02/day-02/diagonal_matrix.py
# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

#for i in range(4):
#    a = [0]
#    a[i] = 1
#    print(a)
a = []
b = []
size = 4
 
for i in range(size):
    b = []
    for j in range(size):
        if i == j: 
            b.append('1')
        else:
            b.append('0')
    a.append(b)

for i in range(size):
     print(a[i])

File: week-07/day-02/diagonal-matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

let size = 4;
let myMatrix = [];

for (let i = 0; i < size; i++) {
  myMatrix.push('0 '.repeat(size - 1) + '1 ' + '0 '.repeat(i));
}

console.log(myMatrix);
